pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go

```diff
@@ -70,7 +70,7 @@ func (j *KubeflowJob) RunWithPodSetsInfo(podSetInfos []jobframework.PodSetInfo)
 }
 
 func (j *KubeflowJob) RestorePodSetsInfo(podSetInfos []jobframework.PodSetInfo) bool {
-	orderedReplicaTypes := j.KFJobControl.OrderedReplicaTypes(j.KFJobControl.ReplicaSpecs())
+	orderedReplicaTypes := j.OrderedReplicaTypes()
 	changed := false
 	for index, info := range podSetInfos {
 		replicaType := orderedReplicaTypes[index]
```

Comment:
we need to check if it exists in the spec here, right?

Reply:
ahh, I see that you are filtering in KubeflowJob

Reply:
Right.